include/dlaf/communication/broadcast_panel.h

```diff
@@ -140,57 +140,22 @@ void broadcast(const comm::Executor& ex, comm::IndexT_MPI rank_root,
 
   const auto& dist = panel.parentDistribution();
 
+  DLAF_ASSERT(square_size(dist), dist.size());  // TODO this can be relaxed
```

Comment:
@rasolca since we can shrink the panel from both ends, I think we should relax this and impose that the sub-matrix described by the two panels is a square matrix (see few lines below `DLAF_ASSERT(panelSize == panelTSize, panelSize, panelTSize)`).

does it sound to you?

Reply:
Not sure if it is something useful.
As transposing is used with symmetric matrices, they are square and the mirroring is always against the diagonal.
I would keep it simple, and extend it only if we have a use case.